dist-persist/wbstack/src/Internal/WbStackPlatformReservedUser.php

```diff
@@ -131,10 +130,31 @@ public static function getOAuthConsumer($consumerName, $version) {
             return false;
         }
 
-        return [
+        $data = [
             'agent' => $c->getName(),
             'consumerKey' => $c->getConsumerKey(),
             'consumerSecret' => \MediaWiki\Extension\OAuth\Backend\Utils::hmacDBSecret( $c->getSecretKey() ),
         ];
+
+        $a = \MediaWiki\Extension\OAuth\Backend\ConsumerAcceptance::newFromUserConsumerWiki(
+            $db,
+            $user->getId(),
+            $c,
+            $c->getWiki(),
+            \MediaWiki\Extension\OAuth\Backend\ConsumerAcceptance::READ_NORMAL,
+            $c->getOAuthVersion(),
+        );
+
+        if ( $a !== false ) {
+            $data['accessKey'] = $a->getAccessToken();
+            $data['accessSecret'] = $a->getAccessSecret();
+        }
+
+        return $data;
+    }
+
+    private static function expandConsumerName(string $consumerName, bool $ownerOnly): string
```

Comment:
This method exists as subsequent lookup happens by name/version and does not consider whether a different "ownerOnly" parameter was passed than what was used on creation.

The approach works, but is definitely not pretty. If someone has a better idea how to solve this I'd be happy to learn about it.

Reply:
I'm not sure it is a "better" solution but would it be possible to instead assert (in the case that no consumer is made) that the ownerOnly parameter is set as requested and error otherwise?

Reply:
Yeah I think I like that better. Will change accordingly.

Reply:
Changed in 376a062